### DISCUSSION TITLE
Alpha release 1.8.0-alpha.0

### DISCUSSION THREAD

**Discussion by tabemann** [tabemann/zeptoforth#136] [General]
This initial alpha release covers the initial additions to support the `rp2350` platform and changes to support the `rp2040` and `rp2040_big` platforms alongside the `rp2350` platform (as significant changes were made to the `rp2040` and `rp2040_big` platforms so as to be able to share code with the `rp2350` platform).

<hr /><em>This discussion was created from the release <a href='https://github.com/tabemann/zeptoforth/releases/tag/v1.8.0-alpha.0'>Alpha release 1.8.0-alpha.0</a>.</em>

**Comment by znmeb** [comment]
It works on a SparkFun Pro Micro RP2350 (Full USB - connected to `/dev/ttyACM0`:

```
Welcome to minicom 2.8

OPTIONS: I18n
Port /dev/ttyACM0, 21:29:52

Press CTRL-A Z for help on special keys


Welcome to zeptoforth
Built for rp2350, version 1.8.0-alpha.0, on Thu Aug 29 07:17:22 PM CDT 2024
zeptoforth comes with ABSOLUTELY NO WARRANTY: for details type `license'
 ok
2 3 + . 5  ok
```

I don't have a test suite - if you have one I'll run it. I also have a Pimoroni Pico Plus 2 (https://shop.pimoroni.com/products/pimoroni-pico-plus-2?variant=42092668289107) and an iLabs Challenger+ RP2350 WiFi6/BLE5 (https://ilabs.se/challenger-rp2350-wifi-ble/). They all have decent amounts of PSRAM but none of them are breadboarded so I can only test USB I/O at the moment. 

I'm going to attempt to get the WiFi / Bluetooth working on the Challenger+ in the next day or so. It's on a separate chip and has its own API.

**Comment by tabemann** [comment]
There is already a known issue with this alpha release -- if you enter
REBOOT to reboot while using the USB CDC console on the RP2350, it is
liable to lock up the USB CDC console so hard that physically disconnecting
and reconnecting your USB cable is needed to resolve the lock-up. This does
not seemingly affect rebooting using control-C or the Reboot button in
zeptocom.js (but I suspect that the same may happen if you use those whole
zeptoforth is writing to the console). The exact issue appears to be with
the MCU being reset while USB is actively sending data, as the MCU reset
logic has changed with zeptoforth on the RP2350. I had tried adding logic
for forcing flushing of data prior to reboot, but that seems to only
currently help with the serial console.

**Comment by tabemann** [comment]
Also note that zeptoforth does not currently support PSRAM, or flash higher
than 4 MB -- these are things that will come later once everything else is
nailed down. (Note that on the RP2350 the first 2 MB of flash is devoted to
the flash dictionary and the second 2 MB is devoted to block/filesystem
storage.)

One other note, which I should report to the RPi people sometime, is that
rebooting the MCU causes the Always-On Timer to lose time such that time
travel into the past occurs! I checked -- it wasn't merely halting the
time, it was actually going backwards in time.

**Comment by tabemann** [comment]
I have made a fix to the USB CDC reboot lockup issue, and am putting
together a new alpha release right now containing this fix.

Travis
